File: web/domain/pondLookup.js
function normalize(value) {
  return value == null ? '' : String(value).trim();
}

export function findByName(envelopes, name) {
  if (!envelopes || name == null) return null;
  const normalized = normalize(name);
  if (normalized === '') return null;
  for (const envelope of envelopes) {
    if (!envelope) continue;
    const envName = envelope.name;
    if (envName != null && normalize(envName) === normalized) {
      return envelope;
    }
  }
  return null;
}

export function canonicalName(envelopes, name) {
  const envelope = findByName(envelopes, name);
  return envelope != null ? envelope.name : null;
}

File: app/src/main/java/com/example/envelopemoney/PondLookup.java
package com.example.envelopemoney;

import androidx.annotation.Nullable;

import java.util.List;

/**
 * Resolves pond names against the live pond list using trimmed equality.
 */
public final class PondLookup {

    private PondLookup() {
    }

    @Nullable
    public static Envelope findByName(@Nullable List<Envelope> envelopes, @Nullable String name) {
        if (envelopes == null || name == null) {
            return null;
        }
        String normalized = normalize(name);
        if (normalized.isEmpty()) {
            return null;
        }
        for (Envelope envelope : envelopes) {
            if (envelope == null) {
                continue;
            }
            String envName = envelope.getName();
            if (envName != null && normalize(envName).equals(normalized)) {
                return envelope;
            }
        }
        return null;
    }

    @Nullable
    public static String canonicalName(@Nullable List<Envelope> envelopes, @Nullable String name) {
        Envelope envelope = findByName(envelopes, name);
        return envelope != null ? envelope.getName() : null;
    }

    private static String normalize(String value) {
        return value.trim();
    }
}
